### ISSUE TITLE
360 Image microsite implementation

### ISSUE THREAD

**Issue by inDigiNeous** [vizorvr/vizor#1073]
- Drag and drop a stereo image file into the browser window
- On mobile, show camera roll opening

- Upon succesfully loading the image, insert the image in the "360 photo template" and automatically publish the graph

This should give the user a 3D photo view immediately in an easy and quick way.

**Comment by inDigiNeous** [comment]
First prototype implemented and live at http://sakari.vizor.io/data/360_index.html


Still need to think about the authentication, as you need to be logged in to publish the graph.

Do we just use some session key that is infinitely logged in, or log in automatically some user ? Would this be the best approach, @kschzt, @mattatgit ?

**Comment by mattatgit** [comment]
Looks like that prototype got blanked by the database crash. Can you put it up again somewhere please (really hope you had a local backup) ?

**Comment by mattatgit** [comment]
Re your question – from a UX perspective, no login is what we need. Just land & go.

**Comment by inDigiNeous** [comment]
@mattatgit It is back up now, you need to login to sakari.vizor.io though first.

**Comment by mattatgit** [comment]
ok cool will check it

**Comment by inDigiNeous** [comment]
We discussed the automatic login implementation details with @kschzt  and came to the conclusion that it is best to implement specific methods used by the site as anonymous versions, so that we don't run into somebody modifying the graphs published by somebody else through this, as there is no easy way to prevent this.

Going to implement anonymous versions of:

- graphController.save
- imageController.save
- assetService.save

This way we can just upload the image anonymously, publish the graph anonymously and keep the completely separate, we might need such functionality any way in the future.

**Comment by inDigiNeous** [comment]
Btw @mattatgit  the sakari.vizor.io is not now running the 360 site branch, so let me know if you want to test that.

**Comment by inDigiNeous** [comment]
Couple of things to maybe take into account before finishing this. Currently the image processing takes quite a while compared to the image uploading, at least with fast connections. 

The current progress bar goes to 50% once the image upload is done. Would have still check:

* Is the core.progress correctly wired to the progress bar? Or should we compensate so that we scale it in the range 50 .. 100 to be in correlation with the 0 .. 50% image loading progress
* Maybe limit the image upload to 40% and then animate between 40-50 slowly while the image processing is happening, to give the illusion that something is happening. Then when the image processing is done, continue from 50 .. 100 with the E2.core progress.

**Comment by mattatgit** [comment]
> Maybe limit the image upload to 40% and then animate between 40-50 slowly while the image processing is happening, to give the illusion that something is happening. Then when the image processing is done, continue from 50 .. 100 with the E2.core progress.

Sounds like a good idea – nothing worse than a progress bar that seems to be hanging or doing nothing

**Comment by gmarinov** [comment]
copying Matt's comment from #1136

> Atomic prototype for desktop behaviour
> https://app.atomic.io/d/8DAbhLLuacLK
> Sketch file name (Sketch 3.5.1)
> 360_Microsite_160128_s

**Comment by mattatgit** [comment]
Repasting from flowdock

My feedback :

* can we have a 360 photo on top page to test with please (one below)
* it shows default textures in the load process
* need a nicer "share" dialogue, this one not looking too good
* the weird pause in the loader bar makes it look like it's broken - can we do something here ?
* the UI on title screen shouldn’t disappear on a timer
* animation on icons to return
![great wall of china sacha arnoud](https://cloud.githubusercontent.com/assets/6858525/13217904/06349a1e-d9ba-11e5-93c8-31ac7d78ba93.jpg)

**Comment by mattatgit** [comment]
We need to not have the entire site UI disappear on a timer, that sort of destroys the whole thing. Can't consider it done until that is fixed. Also just wondering – it's up on 360.vizor.io because staging ? Or is there some opposition to putting it up at the 360vr.io URL ?
